filesystem/usr/local/share/landoop/etc/supervisord.d/99-supervisord-running-sample-data.conf

```diff
@@ -78,10 +78,10 @@ stdout_logfile=/var/log/lenses-processor.log
 autorestart=false
 startretries=1
 
-[program:lenses-policy]
+[program:lenses-demo-setup]
 user=nobody
-command=bash -c 'for ((i=0;i<90;i++)); do sleep 2; curl "http://localhost:$LENSES_PORT" | grep "Lenses" && { sleep 10; break; }; done; /usr/local/share/landoop/sample-data/lenses-policy.sh'
+command=bash -c '/usr/local/share/landoop/wait-scripts/wait-for-lenses.sh; /usr/local/share/landoop/sample-data/lenses-demo-setup.sh'
```

Comment:
You need `eval` and the script reference:
```
command=bash -c 'eval $WAIT_SCRIPT_LENSES; /usr/local/share/landoop/sample-data/lenses-demo-setup.sh'
```
Currently, in `setup-and-run.sh` we use the registry script for the `WAIT_SCRIPT_LENSES` (line 86). Feel free to change it to your version.

Reply:
Changed for lenses-demo-setup.sh. Tried to make change it in `setup-and-run.sh` as well, but apparently we use `$WAIT_SCRIPT_LENSES` for `10-lenses-conf` as well which caused problems. To remedy that we'd have to make the change you proposed and also change `10-lenses-conf` to $WAIT_SCRIPT_REGISTTRY . Shall I do that?

Reply:
Nice catch, you are correct!